src/components/KEmptyState/KEmptyState.vue

```diff
@@ -32,31 +37,67 @@
           </p>
         </slot>
       </div>
+      <div
+        v-if="(actionButtonVisible && actionButtonText) || $slots.action"
+        class="empty-state-action"
+      >
+        <slot name="action">
+          <KButton
+            :disabled="actionButtonDisabled"
+            type="button"
+            @click="emit('click-action')"
+          >
+            {{ actionButtonText }}
+          </KButton>
+        </slot>
+      </div>
     </div>
     <div
-      v-if="(actionButtonVisible && actionButtonText) || $slots.action"
-      class="empty-state-action"
+      v-if="features.length"
+      class="empty-state-features-container"
     >
-      <slot name="action">
-        <KButton
-          :disabled="actionButtonDisabled"
-          type="button"
-          @click="emit('click-action')"
-        >
-          {{ actionButtonText }}
-        </KButton>
-      </slot>
+      <template
+        v-for="feature in features"
+        :key="feature"
+      >
+        <KCard class="empty-state-feature-card">
+          <template #title>
+            <div
+              v-if="feature.key && $slots[`feature-${feature.key}-icon`]"
+              aria-hidden="true"
+              class="feature-icon"
+            >
+              <slot :name="`feature-${feature.key}-icon`" />
```

Comment:
Same comment on slot naming, I think the key should be at the end.

Also, what if the key has invalid characters (e.g. space, etc.)?